### PR TITLE
Bug fix/skip ui elements

### PR DESCRIPTION
**Pull request by jordanmaguire**
1. Removed the dependency of ModelConfiguration with YAML
2. Fixed stupid bug in skip_ui args that was causing no views/controllers to be created. Added specs to make sure I don't fuck myself in the face again. Not sure how I missed them initially. Must have been tired.

### CODE REVIEW

File: lib/model_configuration/model_configuration.rb [hunk 1]
```diff
@@ -63,46 +53,33 @@ def skip_ui?
   end
 
   def show_index?
-    @show_index && !skip_ui?
+    !@skip_index && !skip_ui?
```
Comment (swebb) [inline lib/model_configuration/model_configuration.rb]: [pp] ```!(@skip_index || skip_ui?)```

**Comment by swebb** [comment]
:+1:

**Comment by osiro** [comment]
:+1:

File: lib/model_configuration/model_configuration.rb (hunk 1 above)
Comment (jordanmaguire) [inline lib/model_configuration/model_configuration.rb]: I did consider this but I figured this was more consistent with how I'd describe the condition with speech

**Pull request merged**